### PR TITLE
Add draggable highlight edge resizing

### PR DESCRIPTION
**Pull request by ayushdeolasee**
## Summary
- Adds draggable start/end handles for PDF highlight selection so highlighted text can be resized directly in the viewer.
- Reworks PDF selection/highlight plumbing to preserve selection state while supporting smoother edge dragging and safer bounds handling.
- Refactors the app shell around a single document-centric store flow, with updated toolbar, sidebar, and web/PDF viewer wiring.
- Removes the old pane/workspace split-screen implementation and associated tests in favor of the simplified document layout.
- Adds regression coverage for highlight resize behavior, including edge crossing, empty-page handling, and dragging past text bounds.

## Testing
- Ran the updated PDF persistence and highlight resize test coverage in `Tests/PdfPersistenceTests.swift`.
- Verified highlight resizing cases for:
  - dragging the end handle past the text end
  - dragging the end handle past the start
  - dragging into the left margin
  - dragging the start handle left
  - resizing on an empty page
- Not run: full app UI regression pass on macOS.
- Not run: complete build of every removed pane/workspace path, since those files were deleted in this change.

## Summary by CodeRabbit

* **New Features**
  * Added draggable highlight edge-resize handles with live preview and saving on drag end.
  * Automatically re-applies the selected highlight after the PDF view re-initializes.
  * Web highlight resizing now supports persisting updated page info (including virtual page crossings).
* **Bug Fixes**
  * Improved resize stability around bounds (including empty/no-text pages) and kept resize state consistent during selection changes.
  * Highlight creates now appear immediately with optimistic updates, then reconcile with the saved annotation.
* **Tests**
  * Added edge-case coverage for highlight resize behavior (over-dragging and empty pages).

### CODE REVIEW

File: Vellum/Stores/AnnotationStore.swift [hunk 1]
```diff
@@ -186,16 +186,46 @@ final class AnnotationStore {
...
+                if self.selectedAnnotationId == tempId {
+                    self.selectedAnnotationId = nil
+                }
             }
-            return nil
-        } catch {
-            NSLog("[annotation-store] Failed to create \(label): \(error)")
-            return nil
         }
+        return optimistic
```
[the remark this answers is not in the document]
Comment (ayushdeolasee) [reply Vellum/Stores/AnnotationStore.swift:258]: Fixed in 65436f7. `AnnotationStore` now keeps a `pendingCreates` map of tempId → in-flight create task; `updateAnnotation`/`deleteAnnotation` resolve `temp-` ids by awaiting that task and retargeting the persisted id (or dropping the edit if the create failed), so mutations queue behind the create instead of failing against an unresolvable placeholder.

File: Vellum/Views/PDF/PdfSelectionBridge.swift [hunk 2]
```diff
@@ -530,6 +535,134 @@ final class PdfViewerController {
...
+        guard let final, final.rects != annotation.positionData?.rects else { return }
+        Task { [weak self] in
+            await self?.annotationStore?.updateAnnotation(UpdateAnnotationInput(
+                id: annotation.id, color: nil, content: nil, positionData: final))
+        }
+    }
+
+    func cancelHighlightResize() {
+        highlightResize = nil
+    }
```
[the remark this answers is not in the document]
Comment (ayushdeolasee) [reply Vellum/Views/PDF/PdfSelectionBridge.swift:568]: Fixed in 65436f7. `HighlightLayer` now calls `controller.cancelHighlightResize()` whenever `selectedAnnotationId` changes, so a drag that ends without `onEnded` (gesture cancelled by a view rebuild) can no longer leave a stale preview for the next selection of the same highlight.

File: Vellum/Views/Web/WebContentScript.swift [hunk 3]
```diff
@@ -628,6 +656,334 @@ enum WebContentScript {
...
+    post("highlight-resized", {
+      id: state.id,
+      start: start,
+      end: end,
+      text: text,
+      prefix: ctx.prefix,
+      suffix: ctx.suffix,
+      pageNumber: pageForRaw(start),
+    });
+  }
```
[the remark this answers is not in the document]
Comment (ayushdeolasee) [reply Vellum/Views/Web/WebContentScript.swift]: Fixed in 65436f7 as suggested: the guard now runs after `collapseWs(...).trim()` and bails when the resulting quote is empty, so a whitespace-only drag behaves like a no-op instead of persisting an empty `selectedText` that the `hasQuote` filter would drop on the next round-trip. Verified the resize harness suite still passes.

File: Vellum/Views/Web/WebViewerView.swift [hunk 4]
```diff
@@ -737,6 +754,27 @@ final class WebViewerController: NSObject {
...
+                startOffset: start,
+                endOffset: end,
+                prefix: data["prefix"] as? String,
+                suffix: data["suffix"] as? String,
+                viewportOffset: nil)
+            Task {
+                await annotationStore.updateAnnotation(UpdateAnnotationInput(
+                    id: id, color: nil, content: nil, positionData: positionData))
+            }
+
```
[the remark this answers is not in the document]
Comment (ayushdeolasee) [reply Vellum/Views/Web/WebViewerView.swift:780]: Fixed in 65436f7. `UpdateAnnotationInput` gained an optional `pageNumber`, the `highlight-resized` handler now passes the page computed by the content script, and `WebSessionBackend.updateAnnotation` + the optimistic store update apply it. The PDF backend intentionally ignores it (PDF annotations never change pages on resize).
